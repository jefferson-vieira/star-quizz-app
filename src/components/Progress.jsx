import React from 'react';

const Progress = ({ valuenow }) => (
  <div className="progress">
    <div
      className="progress-bar progress-bar-striped progress-bar-animated bg-warning"
      role="progressbar"
      aria-valuenow={valuenow}
      aria-valuemin="0"
      aria-valuemax="100"
      style={{ width: `${valuenow}%` }}
    />
  </div>
);

export default Progress;
