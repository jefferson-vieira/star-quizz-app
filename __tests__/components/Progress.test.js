import React from 'react';
import { shallow, mount } from 'enzyme';

import Progress from '../../src/components/Progress';

describe('Testing Progress Component...', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Progress />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should show progress', () => {
    const wrapper = mount(<Progress />);
    wrapper.setProps({ valuenow: 50 });
    expect(wrapper.find('.progress-bar').prop('style').width).toBe('50%');
  });
});
