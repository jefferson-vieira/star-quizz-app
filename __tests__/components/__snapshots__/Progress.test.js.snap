// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Progress Component... should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Progress />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        aria-valuemax="100"
        aria-valuemin="0"
        className="progress-bar progress-bar-striped progress-bar-animated bg-warning"
        role="progressbar"
        style={
          Object {
            "width": "undefined%",
          }
        }
      />,
      "className": "progress",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-valuemax": "100",
        "aria-valuemin": "0",
        "aria-valuenow": undefined,
        "className": "progress-bar progress-bar-striped progress-bar-animated bg-warning",
        "role": "progressbar",
        "style": Object {
          "width": "undefined%",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          aria-valuemax="100"
          aria-valuemin="0"
          className="progress-bar progress-bar-striped progress-bar-animated bg-warning"
          role="progressbar"
          style={
            Object {
              "width": "undefined%",
            }
          }
        />,
        "className": "progress",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-valuemax": "100",
          "aria-valuemin": "0",
          "aria-valuenow": undefined,
          "className": "progress-bar progress-bar-striped progress-bar-animated bg-warning",
          "role": "progressbar",
          "style": Object {
            "width": "undefined%",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
